integration-libs/punchout/components/punchout-requisition/punchout-requisition.component.ts

```diff
@@ -0,0 +1,91 @@
+/*
+ * SPDX-FileCopyrightText: 2025 SAP Spartacus team <spartacus-team@sap.com>
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import {
+  ChangeDetectionStrategy,
+  Component,
+  ElementRef,
+  inject,
+  OnInit,
+  ViewChild,
+} from '@angular/core';
+import { FormBuilder, FormGroup } from '@angular/forms';
+import { RoutingService } from '@spartacus/core';
+import {
+  PUNCHOUT_ERROR_PAGE_URL,
+  PunchoutFacade,
+  PunchoutRequisition,
+} from '@spartacus/punchout/root';
+import { filter, map, Observable, switchMap, take, tap, timer } from 'rxjs';
+
+@Component({
+  selector: 'cx-punchout-requsition',
+  templateUrl: './punchout-requisition.component.html',
+  changeDetection: ChangeDetectionStrategy.Default,
+  standalone: false,
+})
+export class PunchoutRequisitionComponent implements OnInit {
+  @ViewChild('punchoutFormElement')
+  punchoutFormElement!: ElementRef<HTMLFormElement>;
+  protected punchoutFacade = inject(PunchoutFacade);
+  protected routingService = inject(RoutingService);
+  punchoutFormGroup: FormGroup;
+
+  protected formBuilder = inject(FormBuilder);
+
+  punchoutRequisition$: Observable<PunchoutRequisition | undefined> =
+    this.punchoutFacade.getPunchoutSessionRequisition().pipe(
+      take(1),
+      tap((punchoutRequisition) => {
+        this.listenAndSubmitForm(punchoutRequisition).subscribe({
+          next: () => {
+            this.punchoutFormElement.nativeElement.submit();
```

Comment:
It is the only way to submit a POST form with 'action' attribute programmatically.
As long as nativeElement is not being modified, it is safe.